gc/base/ObjectScanner.hpp

```diff
@@ -61,7 +68,8 @@ class GC_ObjectScanner : public MM_BaseVirtual
 	uintptr_t _leafMap;						/**< Bit map of reference slots in object that refernce leaf objects */
 #endif /* defined(OMR_GC_LEAF_BITS) */
 	fomrobject_t *_scanPtr;					/**< Pointer to base of object slots mapped by current _scanMap */
-	GC_SlotObject _slotObject;				/**< Create own SlotObject class to provide output */
+	GC_SlotObject *_slotObjectPtr;			/**< Points to effective slot object for receiving refernce slots */
+	GC_SlotObject _slotObject;				/**< Local GC_SlotObject instance to provide default for _slotObjectPtr */
```

Comment:
Why is there an embedded SlotObject if users are expected to set a pointer?

Reply:
That is to maintain backwards compatibility with existing object scanner implementations and use, as well as a default option for use cases to do not require/want to use their own slot object. The default constructor sets the new _slotObjectPtr to point to this instance, the implementation now performs all slot operations using _slotObjectptr. 

This commit adds 8 bytes to the size of the GC_SlotObject. In future we might want to add a GC_SlotObject * parameter to the default GC_ObjectScanner constructor. In that case we can eliminate the GC_SlotObject instance (8 or 16 bytes, compressed or uncompressed). 

I expect GC_ObjectScanner subclasses will not be concerned with this. This change is being introduced to allow GC components that use object scanners to specify specialized slot object instances. This will allow using components to include fields in the slot object for values (eg, object size) that would otherwise need to be calculated >1 times. Eg, component needs object size in >1 context but must calculate it each time because there is no place to cache result of first calculation.